src/test/java/com/programmers/commandline/domain/consumer/repository/impl/ConsumerNamedJdbcRepositoryTest.java

```diff
@@ -0,0 +1,165 @@
+
+        //then
+        assertThat(updateConsumer.getName(), is(updateName));
+        assertThat(updateConsumer.getEmail(), is(updateEmail));
+    }
+
+    @Test
+    @DisplayName("리스트의 소비자를 전부 저장하고 저장된 소비자의 갯수를 count 매서드로 반환 받아서 리스트의 길이와 검증하라")
+    void count() {
+        //given
+        List<Consumer> consumers = new ArrayList<>();
+        Consumer consumer = new Consumer(
+                UUID.randomUUID(),
+                "test_user",
+                "test_user@naver.com",
+                LocalDateTime.now());
+        consumers.add(consumer);
+        //when
+        consumers.forEach(consumerInList -> {
+            consumerNamedJdbcRepository.insert(consumerInList);
+        });
+
+        int count = consumerNamedJdbcRepository.count();
+
+        //then
+        assertThat(count, is(consumers.size()));
+    }
+
+    @Test
+    @DisplayName("소비자를 저장하고 모든 소비자를 찾습니다. 그리고 소비자의 ID를 검증하라")
+    void findAll() {
+        //given
+        Consumer consumer = new Consumer(
+                UUID.randomUUID(),
+                "test_user",
+                "test_user@naver.com",
+                LocalDateTime.now());
+
+        //when
+        consumerNamedJdbcRepository.insert(consumer);
+        List<Consumer> consumers = consumerNamedJdbcRepository.findAll();
+
+        //then
+        consumers.forEach(foundConsumer -> {assertThat(foundConsumer.getConsumerId(),is(foundConsumer.getConsumerId()));});
+    }
+
+    @Test
+    @DisplayName("소비자를 저장하고 저장할 때 사용한 id값을 가지고 조회하고 검증하라")
+    void findById() {
+        //given
+        UUID uuid = UUID.randomUUID();
+        Consumer consumer = new Consumer(uuid, "test_user", "test_user@naver.com", LocalDateTime.now());
+
+        //when
+        consumerNamedJdbcRepository.insert(consumer);
+        Optional<Consumer> foundConsumer = consumerNamedJdbcRepository.findById(uuid.toString());
+
+        //then
+        assertThat(foundConsumer.get().getConsumerId(), is(uuid));
+    }
+
+    @Test
+    @DisplayName("소비자를 저장하고 저장할 때 사용한 name값을 가지고 조회하라")
+    void findByName() {
+        //given
+        String name = "test_user";
+        Consumer consumer = new Consumer(UUID.randomUUID(), name, "test_user@naver.com", LocalDateTime.now());
+
+        //when
+        consumerNamedJdbcRepository.insert(consumer);
+        Optional<Consumer> foundConsumer = consumerNamedJdbcRepository.findByName(name);
+
+        //then
+        assertThat(foundConsumer.get().getName(), is(name));
+    }
+
+    @Test
+    @DisplayName("소비자를 저장하고 저장할 때 사용한 email값을 가지고 조회하라")
+    void findByEmail() {
+        //given
+        String email = "test_user@naver.com";
+        Consumer consumer = new Consumer(UUID.randomUUID(), "test_user", email, LocalDateTime.now());
+
+        //when
+        consumerNamedJdbcRepository.insert(consumer);
+        Optional<Consumer> foundConsumer = consumerNamedJdbcRepository.findByEmail(email);
+
+        //then
+        assertThat(foundConsumer.get().getEmail(), is(email));
```

Comment:
```suggestion
        Consumer foundConsumer = consumerNamedJdbcRepository.findByEmail(email).get();

        //then
        assertThat(foundConsumer.getEmail(), is(email));
```
주관적인 내용이지만 다음과 같이 작성해보면 어떨지 :9
foundConsumer 이니 optional 타입보다는 바로 consumer 타입으로 메서드를 이용하는 것이 가독성이 더 좋다고 생각합니당 😆

Reply:
서브멘톤님의견에 동의 합니다. Test 환경에서는 굳이 Optionel을 사용할 필요성을 느끼지 못하였습니다.